ui/perfherder/compare/CompareSubtestsView.jsx

```diff
@@ -0,0 +1,253 @@
+
+import withValidation from './Validation';
+import CompareTableView from './CompareTableView';
+
+// TODO remove $stateParams and $state after switching to react router
+export class CompareSubtestsView extends React.PureComponent {
+  createQueryParams = (parent_signature, repository, framework) => ({
+    parent_signature,
+    framework,
+    repository,
+  });
+
+  getQueryParams = (timeRange, framework) => {
+    const {
+      originalProject,
+      newProject,
+      originalRevision,
+      newRevision,
+      newResultSet,
+      originalSignature,
+      newSignature,
+    } = this.props.validated;
+
+    const originalParams = this.createQueryParams(
+      originalSignature,
+      originalProject,
+      framework.id,
+    );
+
+    if (originalRevision) {
+      originalParams.revision = originalRevision;
+    } else {
+      // can create a helper function for both views
+      const startDateMs =
+        (newResultSet.push_timestamp - timeRange.value) * 1000;
+      const endDateMs = newResultSet.push_timestamp * 1000;
+      originalParams.startday = new Date(startDateMs)
+        .toISOString()
+        .slice(0, -5);
+      originalParams.endday = new Date(endDateMs).toISOString().slice(0, -5);
+    }
+
+    const newParams = this.createQueryParams(
+      newSignature,
+      newProject,
+      framework.id,
+    );
+    newParams.revision = newRevision;
+    return [originalParams, newParams];
+  };
+
+  createLinks = (oldResults, newResults, timeRange, framework) => {
+    const {
+      originalProject,
+      newProject,
+      originalRevision,
+      newRevision,
+    } = this.props.validated;
+    let links = [];
+
+    if (framework.name === 'talos' && originalRevision) {
+      const params = {
+        originalProject,
+        newProject,
+        originalRevision,
+        newRevision,
+        originalSubtestSignature: oldResults ? oldResults.signature_id : null,
+        newSubtestSignature: newResults ? newResults.signature_id : null,
+      };
+
+      links.push({
+        title: 'replicates',
+        href: `perf.html#/comparesubtestdistribution${createQueryParams(
+          params,
+        )}`,
+      });
+    }
+    const signature_hash = !oldResults
+      ? newResults.signature_hash
+      : oldResults.signature_hash;
+
+    links = createGraphsLinks(
+      this.props.validated,
+      links,
+      framework,
+      timeRange,
+      signature_hash,
+    );
+    return links;
+  };
+
+  getDisplayResults = (origResultsMap, newResultsMap, state) => {
+    const { originalSignature, newSignature } = this.props.validated;
+
+    const { tableNames, rowNames, framework, timeRange } = state;
+    const testsWithNoise = [];
+    let compareResults = new Map();
+    const parentTestName = tableNames[0];
+
+    const oldStddevVariance = {
+      values: [],
+      lowerIsBetter: true,
+      frameworkID: framework.id,
+    };
+    const newStddevVariance = {
```

Comment:
could maybe save a bit of code by doing somethig like:

```js
const newStddevVarious = Object.assign({}, oldStddevVariance);
```

Reply:
True. Or `const newStddevVarious = {...oldStddevVariance}`;